bvm/ballerina-runtime/src/main/java/org/ballerinalang/jvm/MapUtils.java

```diff
@@ -148,9 +151,53 @@ public static void checkIsMapOnlyOperation(BType mapType, String op) {
         switch (mapType.getTag()) {
             case TypeTags.MAP_TAG:
             case TypeTags.JSON_TAG:
+            case TypeTags.RECORD_TYPE_TAG:
                 return;
             default:
                 throw createOpNotSupportedError(mapType, op);
         }
     }
+
+    public static void checkValidFieldForRecord(MapValue<?, ?> m, String  k, String  op) {
+        BType type = m.getType();
+        switch (type.getTag()) {
+            case TypeTags.RECORD_TYPE_TAG:
+                if (!m.containsKey(k)) {
+                    return;
+                }
+                if (isRequiredField((BRecordType) type, k)) {
+                    throw createOpNotSupportedErrorForRecord(type, op);
+                }
+                return;
+            default:
+                return;
+        }
```

Comment:
The only check we need here is the following right, since we aren't doing anything for the other cases.
```ballerina
if (type.getTag() == TypeTags.RECORD_TYPE_TAG && isRequiredField((BRecordType) type, k)) {
    throw createOpNotSupportedErrorForRecord(type, op);
}
```

Reply:
Yeah, let's update since `m.containsKey(k)` will be true anyway for a required field.

Reply:
Fixed